src/ansys/fluent/core/launcher/launcher.py

```diff
@@ -615,20 +611,68 @@ def launch_fluent(
 
     del start_container
 
-    if start_watchdog is None and cleanup_on_exit:
+    if additional_arguments is None:
+        additional_arguments = ""
+    elif fluent_launch_mode == LaunchMode.PIM:
+        raise ValueError(
+            "'additional_arguments' option for 'launch_fluent' is currently not supported "
+            "when starting a remote Fluent PyPIM client."
+        )
+
+    if fluent_launch_mode == LaunchMode.PIM and start_watchdog:
+        raise ValueError(
+            "'start_watchdog' argument for 'launch_fluent' is currently not supported "
+            "when starting a remote Fluent PyPIM client."
+        )
+
+    if (
+        start_watchdog is None
+        and cleanup_on_exit
+        and (
+            fluent_launch_mode == LaunchMode.CONTAINER
+            or fluent_launch_mode == LaunchMode.STANDALONE
+        )
+    ):
         start_watchdog = True
 
-    if dry_run:
-        if not fluent_launch_mode == LaunchMode.CONTAINER:
+    if dry_run and fluent_launch_mode != LaunchMode.CONTAINER:
+        raise ValueError(
+            "'dry_run' argument for 'launch_fluent' currently is only "
+            "supported when starting containers."
+        )
+
+    argvals = locals()
+
+    if fluent_launch_mode != LaunchMode.STANDALONE:
```

Comment:
We can really see how the launcher factory is going to smash this! I will create a detailed gh issue.